controllers/oauth2.js

```diff
@@ -87,26 +87,34 @@ router.get('/consent', csrfProtection, auth.authChecker, (r, w) => {
 router.post('/consent', auth.authChecker, (r, w) => acceptConsent(r, w, r.query.challenge, r.body.grantScopes, true));
 
 router.get('/username/:pseudonym', (req, res, next) => {
-	api(req).get('/pseudonym', {
-		qs: {
-			pseudonym: req.params.pseudonym,
-		},
-	}).then((pseudonym) => {
-		let shortName;
-		let completeName;
-		const anonymousName = '???';
-		completeName = anonymousName;
-		shortName = completeName;
-		if (pseudonym.data.length) {
-			completeName = `${pseudonym.data[0].user.firstName} ${pseudonym.data[0].user.lastName}`;
-			shortName = `${pseudonym.data[0].user.firstName} ${pseudonym.data[0].user.lastName.charAt(0)}.`;
-		}
+	if (req.cookies.jwt) {
+		api(req).get('/pseudonym', {
+			qs: {
+				pseudonym: req.params.pseudonym,
+			},
+		}).then((pseudonym) => {
+			let shortName;
+			let completeName;
+			const anonymousName = '???';
+			completeName = anonymousName;
+			shortName = completeName;
+			if (pseudonym.data.length) {
+				completeName = `${pseudonym.data[0].user.firstName} ${pseudonym.data[0].user.lastName}`;
+				shortName = `${pseudonym.data[0].user.firstName} ${pseudonym.data[0].user.lastName.charAt(0)}.`;
+			}
```

Comment:
what are known error cases, i would not report empty shortName/completeName and expect pseudonym.data exists. the api().then...catch(next) is missing to handle api errors.